parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -1093,10 +1099,38 @@ private ColumnChunkPageReadStore internalReadFilteredRowGroup(BlockMetaData bloc
         }
       }
     }
-    // actually read all the chunks
+    // Vectored IO up.
+
+    List<FileRange> ranges = new ArrayList<>();
     for (ConsecutivePartList consecutiveChunks : allParts) {
-      consecutiveChunks.readAll(f, builder);
+      ranges.add(FileRange.createFileRange(consecutiveChunks.offset, (int) consecutiveChunks.length));
```

Comment:
I would do this the way you were planning to do it initially ( or so it appears). Move this into a readAllVectored method.

Reply:
And make it configurable to choose between vectored I/O and non vectored I/O (see `HadoopReadOptions` and `ParquetReadOptions`)

Reply:
I have changed both the places where I thought the integration can be done. I am not really sure which will give better performance results, which is why left the other portion commented. 
1. One option is to change in readAllVectored
2. Another change (current one) is at the top layer. 
The reason I moved from 1st to 2nd of because of the name ConsecutivePartList. The name suggests it is a consecutive part essentially meaning just a single range and for which we won't be getting the real vectored io benefit like parallel IO and range coalescing.

Reply:
Sure moving to a new method makes sense. will do.